plugins/common/cookie/cookie.go

```diff
@@ -0,0 +1,91 @@
+package cookie
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/cookiejar"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf/config"
+)
+
+type CookieAuthConfig struct {
+	URL    string `toml:"cookie_auth_url"`
+	Method string `toml:"cookie_auth_method"`
+
+	// HTTP Basic Auth Credentials
+	Username string `toml:"cookie_auth_username"`
+	Password string `toml:"cookie_auth_password"`
+
+	Body    string          `toml:"cookie_auth_body"`
+	Renewal config.Duration `toml:"cookie_auth_renewal"`
+
+	client *http.Client
+}
+
+func (c *CookieAuthConfig) Start(client *http.Client) (err error) {
+	c.client = client
+
+	c.setDefaults()
+
+	// add cookie jar to HTTP client
+	if c.client.Jar, err = cookiejar.New(nil); err != nil {
+		return err
+	}
+
+	// continual auth renewal
+	ticker := time.NewTicker(time.Duration(c.Renewal))
+	go func() {
+		for range ticker.C {
+			_ = c.auth()
```

Comment:
The only issue I see here is that we cannot see errors produced in `auth`. Do you have a good idea on how to pass the error on to the caller? Alternatively you might at least want to log the errors...

Reply:
I honestly don't have a good solution.... The telegraf logger instance isn't passed into any of the plugins, so I guess it's up to each plugin to do their own logging?

I can just do a raw log with `log`.

Reply:
Well the correct solution would be to add a 
```golang
Log telegraf.Logger `toml:"-"`
```
line to the struct and pass the parents logger. However, with this solution we need to care for the case where the parent does not set this `Log` (e.g. due to the fact that it has no logger itself).

Alternatively, you could create an `error` channel that the caller can subscribe to. Would that make sense?